### PR TITLE
MediaPreviewer : clean files and manage bitmap with Glide

### PR DESCRIPTION
**Pull request by ganfra**
Some improvements over the MediaPreviewer classes

### CODE REVIEW

**Comment by bmarty** [comment]
```
Search for forbidden patterns in code...
❌ Error: '.{161}' detected 1 time, in file './vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java'.
Rule: Line length is limited to 160 chars. Please split long lines
1 forbidden pattern detected
```

File: vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java [hunk 1]
```diff
@@ -27,187 +28,126 @@
...
-            window.setFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS,
-                    WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
-
-            int statusBarHeight = getStatusBarHeight();
-
-            statusBar.getLayoutParams().height = statusBarHeight;
-            statusBar.setBackgroundColor(color);
-        }
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    private void setStatusBarColor(@ColorRes int statusBarColor) {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java]: Remove then?

File: vector/src/main/java/im/vector/adapters/MediaPreviewAdapter.java [hunk 2]
```diff
@@ -1,99 +1,80 @@
...
 
     /**
      * Initialises the adapter and sets member fields.
+     *
      * @param imagePreviewList list with RoomMediaMessages to be displayed.
-     * @param itemPositionChangedListener listener that listens for clicks on the items.
+     * @param mEventListener
      */
-    public MediaPreviewAdapter(List<RoomMediaMessage> imagePreviewList, ItemPositionChangedListener itemPositionChangedListener) {
+    public MediaPreviewAdapter(final List<RoomMediaMessage> imagePreviewList, EventListener mEventListener) {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/adapters/MediaPreviewAdapter.java]: param name convention

File: vector/src/main/java/im/vector/adapters/MediaPreviewAdapter.java [hunk 3]
```diff
@@ -1,99 +1,80 @@
...
-     */
-    public static class MediaItemViewHolder extends RecyclerView.ViewHolder {
-        @BindView(R.id.image_preview)
-        ImageView mImagePreview;
-
-        MediaItemViewHolder(View itemView) {
-            super(itemView);
-            ButterKnife.bind(this, itemView);
-        }
+        this.mEventListener = mEventListener;
```
Comment (bmarty) [inline vector/src/main/java/im/vector/adapters/MediaPreviewAdapter.java]: remove `this.`

File: vector/src/main/res/layout/activity_media_previewer.xml [hunk 4]
```diff
@@ -1,80 +1,80 @@
...
         android:background="@color/transparent_dark"
-        android:padding="5dip"/>
+        android:padding="8dp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent" />
 
     <TextView
-        android:id="@+id/file_name"
+        android:id="@+id/media_previewer_file_name"
```
Comment (bmarty) [inline vector/src/main/res/layout/activity_media_previewer.xml]: No need for a background here? (I've not run it)

File: vector/src/main/java/im/vector/adapters/MediaPreviewAdapter.java [hunk 5]
```diff
@@ -102,4 +83,25 @@ public void onClick(View view) {
...
+        MediaItemViewHolder(View itemView) {
+            super(itemView);
+            ButterKnife.bind(this, itemView);
+        }
+    }
+
+    public interface EventListener {
+        void onMediaMessagePreviewClicked(RoomMediaMessage roomMediaMessage);
+    }
+
```
Comment (bmarty) [inline vector/src/main/java/im/vector/adapters/MediaPreviewAdapter.java]: remove extra new lines pls

**Comment by bmarty** [comment]
no more crashes when selecting a large number of media?

File: vector/src/main/res/layout/activity_media_previewer.xml (hunk 4 above)
Comment (bmarty) [reply vector/src/main/res/layout/activity_media_previewer.xml]: the text is hardly visible, maybe add transparent background

File: vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java [hunk 6]
```diff
@@ -27,187 +28,126 @@
...
-            mSharedDataItems.add(new RoomMediaMessage(Uri.parse(getIntent().getStringExtra(EXTRA_CAMERA_PICTURE_URI))));
+        final List<RoomMediaMessage> sharedDataItems = new ArrayList<>(RoomMediaMessage.listRoomMediaMessages(getIntent(), RoomMediaMessage.class.getClassLoader()));
+        if (sharedDataItems.isEmpty()) {
+            sharedDataItems.add(new RoomMediaMessage(Uri.parse(getIntent().getStringExtra(EXTRA_CAMERA_PICTURE_URI))));
+        } else {
+            final LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false);
+            mPreviewerRecyclerView.setLayoutManager(linearLayoutManager);
+            final MediaPreviewAdapter mediaPreviewAdapter = new MediaPreviewAdapter(sharedDataItems, this);
+            mPreviewerRecyclerView.setAdapter(mediaPreviewAdapter);
+            updatePreview(sharedDataItems.get(0));
```
Comment (bmarty) [inline vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java]: It's strange, it's working, but I think this line should be outside the else block, no?

Comment (bmarty) [reply vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java]: Actually it's not working if you select "Use native camera" in the settings. Can you check this case pls?

Comment (ganfra) [reply vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java]: Yes, just saw that too

File: vector/src/main/res/layout/activity_media_previewer.xml (hunk 4 above)
Comment (ganfra) [reply vector/src/main/res/layout/activity_media_previewer.xml]: Yep, added.

File: vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java (hunk 1 above)
Comment (ganfra) [reply vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java]: ooops

File: vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java [hunk 7]
```diff
@@ -82,44 +83,45 @@ public boolean onTouch(View v, MotionEvent event) {
...
+            final Uri roomMediaUri = Uri.parse(getIntent().getStringExtra(EXTRA_CAMERA_PICTURE_URI));
+            final RoomMediaMessage roomMediaMessage = new RoomMediaMessage(roomMediaUri);
+            sharedDataItems.add(roomMediaMessage);
+        }
+        final LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false);
+        mPreviewerRecyclerView.setLayoutManager(linearLayoutManager);
+        final MediaPreviewAdapter mediaPreviewAdapter = new MediaPreviewAdapter(sharedDataItems, this);
+        mPreviewerRecyclerView.setAdapter(mediaPreviewAdapter);
+
+        if (!sharedDataItems.isEmpty()) {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java]: I think this cannot be empty, no?

Comment (ganfra) [reply vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java]: Yup

**Pull request merged**